alibi_detect/od/pytorch/ensemble.py

```diff
@@ -18,93 +17,63 @@ def __init__(self):
-        return self.transform(x=x)
+        return self.transform(x)
 
 
 class FitMixinTorch(ABC):
     fitted = False
 
-    def __init__(self):
-        """Fit mixin
-
-        Utility class that provides fitted checks for alibi-detect objects that require to be fit before use.
-        """
-        super().__init__()
-
-    def fit(self, x: torch.Tensor) -> FitMixinTorch:
-        self.fitted = True
-        self._fit(x)
-        return self
+    def fit(self, x: torch.Tensor) -> 'FitMixinTorch':
+        """Public fit method.
 
-    @abstractmethod
-    def _fit(self, x: torch.Tensor):
-        """Fit on `x` tensor.
-
-        This method should be overidden on child classes.
+        The `_fit` method contains the implementation details and should be overridden on child classes. Once the
+        `_fit` method has fit the object the `fitted` attribute should be set to `True`.
 
         Parameters
         ----------
         x
-            Reference `torch.Tensor` for fitting object.
+            `torch.Tensor` to fit object on.
         """
-        pass
+        raise NotImplementedError()
+
+    def set_fitted(self):
+        """Sets the fitted attribute to True.
+
+        Should be called within each transform method.
+        """
+        self.fitted = True
+        return self
 
-    @torch.jit.unused
     def check_fitted(self):
-        """Raises error if parent object instance has not been fit.
+        """Checks to make sure object has been fitted.
 
         Raises
         ------
-        NotFitException
+        NotFittedError
             Raised if method called and object has not been fit.
         """
-        if not self.fitted:
-            raise NotFitException(f'{self.__class__.__name__} has not been fit!')
-
-
-class BaseFittedTransformTorch(BaseTransformTorch, FitMixinTorch):
-    def __init__(self):
-        """Base Fitted Transform class.
-
-        Extends `BaseTransfrom` with fit functionality. Ensures that transform has been fit prior to
-        applying transform.
-        """
-        BaseTransformTorch.__init__(self)
-        FitMixinTorch.__init__(self)
-
-    def transform(self, x: torch.Tensor) -> torch.Tensor:
-        """Checks to make sure transform has been fitted and then applies trasform to input tensor.
-
-        Parameters
-        ----------
-        x
-            `torch.Tensor` being transformed.
-
-        Returns
-        -------
-        the transformed `torch.Tensor`.
-        """
         if not torch.jit.is_scripting():
-            self.check_fitted()
-        return self._transform(x)
+            self._check_fitted()
 
+    @torch.jit.unused
+    def _check_fitted(self):
+        """Raises error if parent object instance has not been fit."""
+        if not self.fitted:
+            raise NotFittedError(f'{self.__class__.__name__} has not been fit!')
 
-class PValNormalizer(BaseFittedTransformTorch):
+
+class PValNormalizer(BaseTransformTorch, FitMixinTorch):
```

Comment:
Nice idea on simplifying and inheriting directly from the base and the mixin separately, think it makes everything more clear!